coderd/coderdtest/testjar/cookiejar.go

```diff
@@ -0,0 +1,33 @@
+package testjar
+
+import (
+	"net/http"
+	"net/url"
+	"sync"
+)
+
+func New() *Jar {
+	return &Jar{}
+}
+
+// Jar exists because 'cookiejar.New()' strips many of the http.Cookie fields
+// that are needed to assert. Such as 'Secure' and 'SameSite'.
+type Jar struct {
+	m      sync.Mutex
+	perURL map[string][]*http.Cookie
+}
+
+func (j *Jar) SetCookies(u *url.URL, cookies []*http.Cookie) {
```

Comment:
I see that this is the same API as net/http/cookiejar, but it's kind of lame that they didn't have this be

```suggestion
func (j *Jar) SetCookie(u *url.URL, cookies ...*http.Cookie) {
```

just a thought tho, we can leave it as is

Reply:
Good thought, but the interface method definition has to match http.CookieJar

https://github.com/golang/go/blob/master/src/net/http/jar.go#L21

Reply:
ooooh it's an interface, makes sense